client/src/app/components/add-rule/add-rule.component.ts

```diff
@@ -232,7 +232,7 @@ export class AddRuleComponent implements OnInit {
    */
   saveRule() {
     // Add owner
-    this.currentRule.owner = this.authService.currentUser?.id;
+    this.currentRule.owner = this.userService.currentUser?.id;
```

Comment:
I would consider having the user service's `currentUser`property to guarantee a user is returned and throwing an error otherwise. Then you could use it throughout the codebase where ever we assume a logged in user exists rather than conditionally using the user properties. Then inconsistencies would immediately become apparent whenever we wrongly assume a logged-in user.

Something like:

```
get currentUser(): User {
  if(this.user) {
    return this.user;
  } else {
    throw new Error('No logged in user')
  }
}
```